src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java

```diff
@@ -51,41 +262,951 @@
+    forwardingRuleClient.close();
+    globalAddressClient.close();
+    globalForwardingRuleClient.close();
+    globalOperationClient.close();
+    healthCheckClient.close();
+    httpHealthCheckClient.deleteHttpHealthCheck(HTTP_HEALTH_CHECK_NAME);
+    httpHealthCheckClient.close();
+    httpsHealthCheckClient.deleteHttpsHealthCheck(HTTPS_HEALTH_CHECK_NAME);
+    httpsHealthCheckClient.close();
     instanceClient.close();
+    instanceGroupClient.deleteInstanceGroup(INSTANCE_GROUP_NAME);
+    instanceGroupClient.close();
+    instanceTemplateClient.close();
+    instanceGroupManagerClient.close();
+    interconnectAttachmentClient.close();
+    interconnectLocationClient.close();
+    interconnectClient.close();
+    licenseClient.deleteLicense(LICENSE_NAME);
+    licenseClient.close();
+    machineTypeClient.close();
+    networkEndpointGroupClient.deleteNetworkEndpointGroup(NETWORK_ENDPOINT_GROUP_NAME);
+    networkEndpointGroupClient.close();
+    networkClient.close();
+    nodeTemplateClient.close();
+    nodeTypeClient.close();
+    nodeGroupClient.close();
+    projectClient.close();
+    regionAutoscalerClient.close();
+    regionClient.close();
+    regionCommitmentClient.close();
+    regionDiskClient.close();
+    regionBackendServiceClient.close();
+    targetHttpProxyClient.close();
+    targetHttpsProxyClient.close();
+    targetInstanceClient.close();
+    Thread.sleep(30000);
+    targetPoolClient.deleteTargetPool(REGION_TARGET_POOL_NAME);
+    targetPoolClient.close();
+    targetSslProxyClient.close();
+    targetTcpProxyClient.close();
+    targetVpnGatewayClient.close();
+    urlMapClient.close();
+    vpnGatewayClient.close();
+    vpnTunnelClient.close();
+    zoneClient.close();
+    if (storage != null) {
+      boolean wasDeleted = RemoteStorageHelper.forceDelete(storage, BUCKET, 10, TimeUnit.SECONDS);
+      if (!wasDeleted && LOG.isLoggable(Level.WARNING)) {
+        LOG.log(Level.WARNING, "Deletion of bucket {0} timed out, bucket is not empty", BUCKET);
+      }
+    }
+  }
+
+  @Test
+  public void listAcceleratorTypesTest() {
+    AcceleratorTypeClient.ListAcceleratorTypesPagedResponse listAcceleratorTypes =
+        acceleratorTypeClient.listAcceleratorTypes(PROJECT_ZONE_NAME);
+    List<AcceleratorType> acceleratorTypes = Lists.newArrayList(listAcceleratorTypes.iterateAll());
+    assertThat(acceleratorTypes).isNotNull();
+    assertTrue(acceleratorTypes.iterator().hasNext());
+    assertNotNull(acceleratorTypes.iterator().next());
+  }
+
+  @Test
+  public void aggregatedListAcceleratorTypesTest() {
+    AggregatedListAcceleratorTypesPagedResponse aggregatedListAcceleratorTypes =
+        acceleratorTypeClient.aggregatedListAcceleratorTypes(PROJECT_NAME);
+    List<AcceleratorTypesScopedList> typesScopedLists =
+        Lists.newArrayList(aggregatedListAcceleratorTypes.iterateAll());
+    assertThat(typesScopedLists).isNotNull();
+    assertTrue(typesScopedLists.iterator().hasNext());
```

Comment:
Why not directly use iterator inside `while` loop?

Reply:
whats the requirement to used while loop here?

Reply:
Bcoz you are wrapping `aggregatedListAcceleratorTypes.iterateAll()` again with a list. If not while loop then you can go for `foreach` loop,

Reply:
done